homeassistant/components/onvif/config_flow.py

```diff
@@ -142,44 +147,33 @@ async def async_step_device(self, user_input=None):
                 data_schema=vol.Schema({vol.Optional(CONF_HOST): vol.In(names)}),
             )
 
-        return await self.async_step_manual_input()
+        return await self.async_step_configure()
 
-    async def async_step_manual_input(self, user_input=None):
-        """Manual configuration."""
+    async def async_step_configure(self, user_input=None, errors=None):
+        """Device configuration."""
         if user_input:
             self.onvif_config = user_input
-            return await self.async_step_auth()
-
-        return self.async_show_form(
-            step_id="manual_input",
-            data_schema=vol.Schema(
-                {
-                    vol.Required(CONF_NAME): str,
-                    vol.Required(CONF_HOST): str,
-                    vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
-                }
-            ),
-        )
-
-    async def async_step_auth(self, user_input=None):
-        """Username and Password configuration for ONVIF device."""
-        if user_input:
-            self.onvif_config[CONF_USERNAME] = user_input[CONF_USERNAME]
-            self.onvif_config[CONF_PASSWORD] = user_input[CONF_PASSWORD]
             return await self.async_step_profiles()
 
+        def conf(name, default=None):
+            return self.onvif_config.get(name, default)
+
         # Username and Password are optional and default empty
         # due to some cameras not allowing you to change ONVIF user settings.
         # See https://github.com/home-assistant/core/issues/39182
         # and https://github.com/home-assistant/core/issues/35904
         return self.async_show_form(
-            step_id="auth",
+            step_id="configure",
             data_schema=vol.Schema(
                 {
-                    vol.Optional(CONF_USERNAME, default=""): str,
-                    vol.Optional(CONF_PASSWORD, default=""): str,
+                    vol.Required(CONF_NAME, default=conf(CONF_NAME)): str,
+                    vol.Required(CONF_HOST, default=conf(CONF_HOST)): str,
+                    vol.Required(CONF_PORT, default=conf(CONF_PORT, DEFAULT_PORT)): int,
+                    vol.Optional(CONF_USERNAME, default=conf(CONF_USERNAME, "")): str,
+                    vol.Optional(CONF_PASSWORD, default=conf(CONF_PASSWORD, "")): str,
                 }
             ),
+            errors=errors,
         )
 
     async def async_step_profiles(self, user_input=None):
```

Comment:
This shouldn't be a step method. It should be a helper since it doesn't have a form.